api/src/main/java/io/opentelemetry/api/metrics/AsynchronousInstrument.java

```diff
@@ -34,13 +34,26 @@
    * exported then it will never be called.
    *
    * @param callback the callback to be executed before export.
+   * @deprecated Please register the callback while building the instrument. This method will be
+   *     deprecated in 0.13.0.
    */
+  @Deprecated
   void setCallback(Callback<R> callback);
 
   /** Builder class for {@link AsynchronousInstrument}. */
-  interface Builder extends Instrument.Builder {
+  interface Builder<R extends Result> extends Instrument.Builder {
+    /**
+     * Sets a callback that gets executed every collection interval.
+     *
+     * <p>Evaluation is deferred until needed, if this {@code AsynchronousInstrument} metric is not
+     * exported then it will never be called.
+     *
+     * @param callback the callback to be executed before export.
+     */
+    Builder<R> setCallback(Callback<R> callback);
```

Comment:
Moving to Builder's seems like a great change. Just confirming, though not for this PR since it's the existing behavior, is `setCallback` good enough and not `addCallback` (more common in callback-oriented APIs I think)?

Reply:
In this API, you can only associate a single callback, so I think it makes sense to call it `set`.